client/reader/stats.js

```diff
@@ -99,6 +99,23 @@ function getLocation( path ) {
 	return 'unknown';
 }
 
+/**
+ *
+ * @param {Object} eventProperties extra event properties to add
+ * @param {*} pathnameOverride Overwrites location used for determining ui_algo. See notes in
+ * `recordTrack` function docs below for more info.
+ * @param {Object|null} post Optional post object used to build post event props.
+ * @returns new eventProperties object with default reader values added.
+ */
+export function buildReaderTracksEventProps( eventProperties, pathnameOverride, post ) {
```

Comment:
Thought: given that we'll be using this from multiple places, might it make sense to push it into a dedicated file so we don't need to pull in this entire file when we're bundling things up? If we do that, we could also use TS for the new function/file.

Reply:
We could move this function and the required `getLocation` elsewhere, but will that save us anything measurable? Wont all of this file be included anyways since many of the functions here are used all throughout the reader, and our other file isn't a package to be used elsewhere?

Reply:
It may or may not have an immediate impact, but I honestly don't know the intricacies of the Webpack bundling or tree-shaking algorithms to predict that. I do know we can't get any benefits if we lump things together, but it's _possible_ if we split things up. I don't think that's enough of a motivation to do a lot of extra work, but it does clean things up for future code or split that may not depend on all the functions in this single file.

Still, definitely not blocking. 😁